Sources/Subjects/ReplaySubject.swift

```diff
@@ -0,0 +1,171 @@
+
+    private var completion: Subscribers.Completion<Failure>?
+    private var isActive: Bool { completion == nil }
+
+    /// The initialization point for `ReplaySubject`s.
+    /// - Parameter maxBufferSize: The maximum number of value events to buffer and replay to all future subscribers.
+    public init(maxBufferSize: UInt) {
+        self.maxBufferSize = maxBufferSize
+    }
+}
+
+// MARK: - `Subject`
+
+extension ReplaySubject: Subject {
+    public func send(_ value: Output) {
+        guard isActive else { return }
+
+        buffer.append(value)
+
+        if buffer.count > maxBufferSize {
+            buffer.removeFirst()
+        }
+
+        subscriptions.forEach { $0.forwardValueToBuffer(value) }
+    }
+
+    public func send(completion: Subscribers.Completion<Failure>) {
+        guard isActive else { return }
+
+        self.completion = completion
+
+        subscriptions.forEach { $0.forwardCompletionToBuffer(completion) }
+    }
+
+    public func send(subscription: Combine.Subscription) {
+        subscription.request(.unlimited)
+    }
+}
+
+// MARK: - `Publisher`
+
+extension ReplaySubject: Publisher {
+    public typealias Output = Output
+    public typealias Failure = Failure
+
+    public func receive<Subscriber: Combine.Subscriber>(
+        subscriber: Subscriber
+    ) where Failure == Subscriber.Failure, Output == Subscriber.Input {
+        let subscriberIdentifier = subscriber.combineIdentifier
+
+        guard !subscriberIdentifiers.contains(subscriberIdentifier) else {
+            subscriber.receive(subscription: Subscriptions.empty)
+            subscriber.receive(completion: .finished)
+            return
+        }
+
+        let subscription = Subscription(
+            downstream: AnySubscriber(subscriber)
+        ) { [weak self] in
+            guard
+                let self = self,
+                let subscriptionIndex = self.subscriptions
+                    .firstIndex(where: { $0.innerSubscriberIdentifier == subscriberIdentifier })
+                else { return }
+
+            self.subscriberIdentifiers.remove(subscriberIdentifier)
+            self.subscriptions.remove(at: subscriptionIndex)
+        }
+
+        subscriberIdentifiers.insert(subscriberIdentifier)
+        subscriptions.append(subscription)
+
+        subscriber.receive(subscription: subscription)
+        subscription.replay(buffer, completion: completion)
+    }
+}
+
+// MARK: - `ReplaySubject.Subscription`
+
+extension ReplaySubject {
+    final class Subscription<Downstream: Subscriber>
+    where Output == Downstream.Input, Failure == Downstream.Failure {
+        // MARK: - Private
+
+        private var demandBuffer: DemandBuffer<Downstream>?
+        private var cancelHandler: (() -> Void)?
+
+        // MARK: - Internal
+
+        let innerSubscriberIdentifier: CombineIdentifier
+
+        init(
+            downstream: Downstream,
+            cancelHandler: (() -> Void)?
+        ) {
```

Comment:
```suggestion
        init(downstream: Downstream, cancelHandler: (() -> Void)?) {
```

Might also be worth naming cancelationHandler, but up to you